Sources/LiveViewNative/Views/Drawing and Graphics/Color.swift

```diff
@@ -46,23 +46,65 @@ struct Color<R: RootRegistry>: View {
 
 private let colorRegex = try! NSRegularExpression(pattern: "^#[0-9a-f]{6}$", options: .caseInsensitive)
 
-extension SwiftUI.Color.RGBColorSpace: AttributeDecodable {
+extension SwiftUI.Color.RGBColorSpace: AttributeDecodable, Decodable {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let string = try container.decode(String.self)
+        
+        if let rgbColorSpace = Self(string: string) {
+            self = rgbColorSpace
+        } else {
+            throw DecodingError.dataCorrupted(.init(codingPath: container.codingPath, debugDescription: "expected valid value for RGBColorSpace"))
+        }
+    }
+
     public init(from attribute: LiveViewNativeCore.Attribute?) throws {
-        switch attribute?.value {
+        if let string = attribute?.value, let rgbColorSpace = Self(string: string) {
+            self = rgbColorSpace
+        } else {
+            throw AttributeDecodingError.missingAttribute(Self.self)
+        }
+    }
+    
+    init?(string: String) {
+        switch string {
         case "srgb": self = .sRGB
         case "srgb-linear": self = .sRGBLinear
         case "display-p3": self = .displayP3
-        default: throw AttributeDecodingError.missingAttribute(Self.self)
+        default: return nil
         }
     }
 }
 
-extension SwiftUI.Color {
+extension SwiftUI.Color: Decodable {
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        
+        if let rgbColorSpace = try container.decode(SwiftUI.Color.RGBColorSpace?.self, forKey: .rgbColorSpace) {
```

Comment:
For all of these, I think you want `decodeIfPresent` rather than decoding an optional type—the absence of the key should be permitted, but currently this would throw if the key's not present.

Reply:
@shadowfacts Thanks, that makes sense. This should be fixed in b17cc69